File: leetcode/1._Two_Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        h = {}

        for i in range(len(nums)):
            if target - nums[i] in h:
                return [i, h[target - nums[i]]]
            else:
                h[nums[i]] = i
                
from collections import defaultdict 

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        stash = defaultdict(int)

        for i, n in enumerate(nums):
            if target-n in stash:
                return [i, stash[target-n]]
            else:
                stash[n] = i

File: yandex_back_prep_cpp/1. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        map <int,int> h;

        for (int i=0; i < nums.size(); i++) {
            if (h.contains(target - nums[i])) {
                return vector<int>({i, h[target - nums[i]]});
            } else {
                h[nums[i]] = i;
            }
        }

        return {0,0};
        
    }
};